src/SUnitRange.jl

```diff
@@ -0,0 +1,12 @@
+# `Start` and `End` should be `Int`
+struct SUnitRange{Start,End} <: StaticVector{Int}
+end
+
+@pure Size(::SUnitRange{Start, End}) where {Start,End} = Size(End-Start+1)
+
+@pure @propagate_inbounds function getindex(x::SUnitRange{Start,End}, i::Int) where {Start, End}
+    @boundscheck if i < Start || i > End
```

Comment:
So it looks like you've defined a static version of what I'm calling an `IdentityRange` or `IdempotentRange`, see https://github.com/JuliaArrays/IdentityRanges.jl/issues/1 and https://github.com/JuliaLang/julia/pull/21052. Just to check, you're aware that these are non-1 indices arrays you've defined here? If this is intentional, you should definitely define the `indices` function for them.

If that's not what you're intending, you should be checking against `1 <= i <= End - Start + 1` and then return `i + Start - 1`.